src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1133,13 +1133,16 @@ internal void HandleAltSvc(IEnumerable<string> altSvcHeaderValues, TimeSpan? res
                 {
                     var value = (AltSvcHeaderValue?)parsedValue;
 
-                    // 'clear' should be the only value present.
-                    if (value == AltSvcHeaderValue.Clear)
+                    lock (SyncObj)
                     {
-                        ExpireAltSvcAuthority();
-                        Debug.Assert(_authorityExpireTimer != null);
-                        _authorityExpireTimer.Change(Timeout.Infinite, Timeout.Infinite);
-                        break;
+                        // 'clear' should be the only value present.
+                        if (value == AltSvcHeaderValue.Clear)
+                        {
```

Comment:
Why is the `if` and `lock` inversion needed here?

Reply:
You're right, there's no need for that 😅 thanks for catching